### ISSUE TITLE
[BUG] Wrong info text during deletion of an unpublished API proxy

### ISSUE THREAD

**Issue by jschaefer77** [Axway-API-Management-Plus/apim-cli#25]
**API-Mgr. CLI version**
1.0.1

**API-Manager and Service-Pack Version**
7.6.2. SP4

**Expected behavior**
As OrgAdmin, if I delete an existing unpublished API proxy I get the following two lines of info which are wrong, because the API proxy was deleted successfully.
- Successfully updated deleted API
- Actual API has been created and is waiting for an approval by an administrator. You may update the pending API as often as you want before it is finally published.

**Actual behavior**

**Log-Output in Debug**
2014   INFO     UpdateExistingAPI| Waiting for approval ... Successfully updated deleted API: 'Petstore API' 1.0.5 (ID: 61c80aae-d3bd-412b-a3f3-5bd2c9694f42)
2015   INFO      APIImportManager| Actual API has been created and is waiting for an approval by an administrator. You may update the pending API as often as you

**Test case sample**
> Please describe the steps you have done to replicate the issue

**Comment by cwiechmann** [comment]
Hi @jschaefer77, I tried to reproduce the issue, but for me it's working as expected. 

When I try to delete an API as an Org-Admin (not having an Admin-Account at all)
```
C:\temp\apim-cli-1.0.1\samples>apim api get -h api-env -u anna -p anna -wide
------------------------------------------------------------------------
API-Manager CLI: 1.0.1

To report issues or get help, please visit:
https://github.com/Axway-API-Management-Plus/apim-cli
------------------------------------------------------------------------
Module: API - E X P O R T / U T I L S (1.0.1)
------------------------------------------------------------------------
2537   INFO     APIManagerAdapter| Successfully connected to API-Manager (7.7.20200130) on: https://api-env:8075
2746   WARN  rOrganizationAdapter| Using OrgAdmin only to load all organizations.
2795   INFO          APIExportApp| 1 API(s) selected.
+--------------------------------------+---------+------+---------+--------+-------------+----------+----------+-----------------+
| API-Id                               | Path    | Name | Version | V-Host | State       | Security | Policies | Organization    |
+--------------------------------------+---------+------+---------+--------+-------------+----------+----------+-----------------+
| 88fb1882-f145-4afe-939e-b27737398da8 | /api/v1 | B2Bi |   1.1.0 |        | unpublished |   apiKey | None     | API Development |
+--------------------------------------+---------+------+---------+--------+-------------+----------+----------+-----------------+

C:\temp\apim-cli-1.0.1\samples>apim api delete -h api-env -u anna -p anna -id 88fb1882-f145-4afe-939e-b27737398da8
------------------------------------------------------------------------
API-Manager CLI: 1.0.1

To report issues or get help, please visit:
https://github.com/Axway-API-Management-Plus/apim-cli
------------------------------------------------------------------------
Module: API - E X P O R T / U T I L S (1.0.1)
------------------------------------------------------------------------
2628   INFO     APIManagerAdapter| Successfully connected to API-Manager (7.7.20200130) on: https://api-env:8075
2820   WARN  rOrganizationAdapter| Using OrgAdmin only to load all organizations.
2858   INFO          APIExportApp| 1 API(s) selected.
1 selected for deletion.
Do you wish to proceed? (Y/N)Y
Okay, going to delete: 1 API(s)
6556   INFO  APIManagerAPIAdapter| API: B2Bi successfully deleted
Done!
```

When importing an API (unpublished) as an Org-Admin and making updates to it:
```
C:\temp\apim-cli-1.0.1\samples>apim api import -h api-env -u anna -p anna -c basic\minimal-config-api-definition.json
------------------------------------------------------------------------
API-Manager CLI: 1.0.1

To report issues or get help, please visit:
https://github.com/Axway-API-Management-Plus/apim-cli
------------------------------------------------------------------------
Module: API - I M P O R T (1.0.1)
------------------------------------------------------------------------
0      INFO     APIManagerAdapter| Successfully connected to API-Manager (7.7.20200130) on: https://api-env:8075
159    WARN  rOrganizationAdapter| Using OrgAdmin only to load all organizations.
237    INFO  IImportConfigAdapter| Reading API-Definition (Swagger/WSDL) from file: 'C:\temp\apim-cli-1.0.1\samples\petstore.json' (absolute path)
328    INFO      APIImportManager| No existing API found, creating new!
1452   INFO          CreateNewAPI| Create unpublished API: 'Sample Minimal API' 1.0.0 based on Swagger 2.0 specification.
3979   INFO          CreateNewAPI| Successfully created unpublished API: 'Sample Minimal API' 1.0.0 (ID: d558abf0-8bae-4479-ac7a-e6c85a87123a)

C:\temp\apim-cli-1.0.1\samples>apim api import -h api-env -u anna -p anna -c basic\minimal-config-api-definition.json
------------------------------------------------------------------------
API-Manager CLI: 1.0.1

To report issues or get help, please visit:
https://github.com/Axway-API-Management-Plus/apim-cli
------------------------------------------------------------------------
Module: API - I M P O R T (1.0.1)
------------------------------------------------------------------------
0      INFO     APIManagerAdapter| Successfully connected to API-Manager (7.7.20200130) on: https://api-env:8075
145    WARN  rOrganizationAdapter| Using OrgAdmin only to load all organizations.
221    INFO  IImportConfigAdapter| Reading API-Definition (Swagger/WSDL) from file: 'C:\temp\apim-cli-1.0.1\samples\petstore.json' (absolute path)
440    INFO      APISpecification| Detected API-Definition-Filesizes: API-Manager: 21159 vs. Import: 21160
443    INFO      APIImportManager| Recognized the following changes. Potentially Breaking: [apiDefinition] plus Non-Breaking: []
444    INFO      APIImportManager| Update API Strategy: Re-Create API as changes can't be applied to existing API.
445    INFO   RecreateToUpdateAPI| Create new API to update existing: 'Sample Minimal API' (ID: d558abf0-8bae-4479-ac7a-e6c85a87123a)
1260   INFO          CreateNewAPI| Create unpublished API: 'Sample Minimal API' 1.0.0 based on Swagger 2.0 specification.
2617   INFO          CreateNewAPI| Successfully created unpublished API: 'Sample Minimal API' 1.0.0 (ID: 67d25059-5a06-4ccf-b351-f760c50fd8dc)
2618   INFO   RecreateToUpdateAPI| New API successfuly created. Going to delete old API: 'Sample Minimal API' 1.0.0 (ID: d558abf0-8bae-4479-ac7a-e6c85a87123a)
2935   INFO  APIManagerAPIAdapter| API: Sample Minimal API successfully deleted
```

**Comment by jschaefer77** [comment]
Hi @cwiechmann, please add to your api configuration file an application and do the delete again.
Then you see additional info messages are the message ... API successfuly deleted.

2424   INFO      ManageClientApps| All desired applications: [[Petstore Test App]] have already a subscription. Nothing to do.
2424   INFO     UpdateExistingAPI| Waiting for approval ... Successfully updated deleted API: 'Petstore API' 1.0.0 (ID: c9ffc938-6348-44f1-bfc3-89066b46abf3)
2426   INFO      APIImportManager| Actual API has been created and is waiting for an approval by an administrator. You may update the pending API as often as you want before it is finally published.

**Comment by cwiechmann** [comment]
I also tried it with an application the Org-Admin has access to and it works as expected. The message: `Waiting for approval` you see is controlled by the following method:
```java
public boolean isAdminAccountNeeded() throws AppException {
	// Initially set, right after comparing the Desired- with Actual-state
	if(this.isAdminAccountNeeded!=null) return Boolean.parseBoolean(this.isAdminAccountNeeded);
	// If the desired & actual API is state Unpublished - No Admin-Account is needed
	if(getDesiredAPI().getState().equals(API.STATE_UNPUBLISHED) && 
			(getActualAPI()==null || getActualAPI().getState().equals(API.STATE_UNPUBLISHED))) {
		this.isAdminAccountNeeded = "false";
	} else {
		this.isAdminAccountNeeded = "true";
	}
	return Boolean.parseBoolean(this.isAdminAccountNeeded);
}
```
Are you deleting the API manually? 
`apim api delete ....`
or is it part on an update action (e.g. you changed the Swagger-File)? Can you please provide the exacts steps you have executed to see this issue?

**Comment by jschaefer77** [comment]
Hi @cwiechmann, I used the following command
`apim api import -c <path-to-config> -s <my-stage>`
I set the state in the API configuration file to 'deleted'!

**Comment by jschaefer77** [comment]
At least, that was the only was with Swagger Promote in the past.

**Comment by cwiechmann** [comment]
Will be fixed with version 1.1.0
